src/main/kotlin/com/bonitasoft/gradle/maven/settings/MavenSettingsPlugin.kt

```diff
@@ -128,24 +129,24 @@ class MavenSettingsPlugin : Plugin<Project> {
     }
 
     private fun registerMirrors(project: Project) {
+        val mirrorsToAdd = mutableMapOf<String, Action<MavenArtifactRepository>>()
         project.repositories.toList().filter { repo ->
             if (repo.name.equals(ArtifactRepositoryContainer.DEFAULT_MAVEN_LOCAL_REPO_NAME)) {
                 return@filter false
             }
             repo.getMirror(settings)?.also { mirror ->
-                if (!project.repositories.names.contains(mirror.id)) {
-                    project.repositories.maven { repo ->
-                        repo.name = mirror.id
-                        repo.url = URI.create(mirror.url)
-                        project.logger.info("Replaced '${repo.name}' with mirror ${mirror.id} configured in maven's settings.xml")
-                    }
-                } else {
-                    project.logger.info("Replaced '${repo.name}' with mirror ${mirror.id} (already created) configured in maven's settings.xml")
+                project.logger.info("Replaced '${repo.name}' with mirror '${mirror.id}' configured in maven's settings.xml")
```

Comment:
Actuallly, this message is not true anymore, right?

Reply:
It is, when the mirror matches the repository (already present or added using profiles), it replaces it.
The bug was that, when a repo with name `my.repo` was present:
* If the mirror with the same id `my.repo` matches the repository
* We were not adding the mirror because one repository was already present with that name
* But then, we were removing the repository

Reply:
Ok, thx for the explanations. I better understand now.